crates/qt-build-utils/src/builder/qresource.rs

```diff
@@ -0,0 +1,213 @@
+// SPDX-FileCopyrightText: 2025 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::{
+    io,
+    path::{Path, PathBuf},
+};
+
+/// An individial `<file>` line within a [QResource]
+pub struct QResourceFile {
+    alias: Option<String>,
+    // TODO: compression
+    // TODO: empty
+    path: PathBuf,
+}
+
+impl<T: AsRef<Path>> From<T> for QResourceFile {
+    fn from(value: T) -> Self {
+        Self::new(value)
+    }
+}
+
+impl QResourceFile {
+    /// Construct a [QResourceFile]
+    pub fn new(path: impl AsRef<Path>) -> Self {
+        Self {
+            alias: None,
+            path: path.as_ref().to_path_buf(),
+        }
+    }
+
+    /// Specify an alias for the [QResourceFile]
+    pub fn alias(mut self, alias: impl Into<String>) -> Self {
+        self.alias = Some(alias.into());
+        self
+    }
+
+    fn write(self, writer: &mut impl io::Write) -> io::Result<()> {
+        let alias = self
+            .alias
+            .map(|alias| format!(" alias=\"{}\"", alias.escape_default()))
+            .unwrap_or_default();
+        let path = self.path.to_string_lossy();
+        write!(writer, "<file{alias}>{path}</file>")
+    }
+}
+
+/// A `<qresource>` block within a [QResources]
+pub struct QResource {
+    language: Option<String>,
+    prefix: Option<String>,
+    files: Vec<QResourceFile>,
+}
+
+impl Default for QResource {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<T: Into<QResourceFile>> From<T> for QResource {
+    fn from(value: T) -> Self {
+        Self::new().file(value)
+    }
+}
```

Comment:
I would like to see a similar implementation for `QResources` itself.

Most likely we want a `From` impl for `T: Into<QResourceFile>`, as well as `T: Iterator<Item=I>, I: Into QResourceFile`,
That way we could have an API on CxxQtBuilder that is simply:

```rust
.resources([my_image.png, my_other_image.png])
```
Which would boil down to a QResources, with a single QResources and two QResourceFile instances.

Reply:
done